### ISSUE TITLE
Serial port nullmodem server only works for a single connect

### ISSUE THREAD

**Issue by ThomasKjoernes** [joncampbell123/dosbox-x#4094]
### Describe the bug

I configure serial port 1 like this:

serial1 = nullmodem server port:5001

Then I successfully connect from the host computer using:

telnet localhost 5001

I can read and write to the port just fine. When closing telnet, DosBox-X will not allow me to connect to again without restarting or using the builtin 'serial' command to set up the port again.

I would expect the server to recover and start listening for another connection once the remote has disconnected.

### Steps to reproduce the behaviour

1) Configure serial port in the conf file (or use the serial command):
serial1 = nullmodem server port:5001

2) Connect from the host computer using telnet:

telnet localhost 5001

3) Disconnect from telnet.

4) Try to connect again as in step 2. It will not connect. 

### Expected behavior

I would expect the server to recover and start listening for another connection once the remote has disconnected.

### What operating system(s) this bug have occurred on?

Windows 10 22H2

### What version(s) of DOSBox-X have this bug?

2022.12.26 (64-bit SDL2)

### Have you checked that no similar bug report(s) exist?

- [X] I have searched and didn't find any similar bug report.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

**Comment by ThomasKjoernes** [comment]
I think I have found the bug. Inside the file: nullmodem.cpp

The method CNullModem::Disconnect is never called. 
It looks like it was supposed to be called inside CNullModem::doReceive where it checks for a return value of -2 from readChar() but this methods will only return -1 indicating an error.

// Adding the following inside readChar seems to work:
// After the line:
// SocketState state = clientsocket->GetcharNonBlock(val);
//
// I added:

    if (state == SocketState::Closed)
        return -2;
